### DISCUSSION TITLE
Resolve() function in 6-cell_counting notebook

### DISCUSSION THREAD

**Discussion by carmenrt** [DeepTrackAI/DeepTrack2#190] [Q&A]
Hi, I am working with the notebook called 6-cell_counting. After generating the predictions after training the model, I want to create a CSV containing the names of the images according to the order of test_paths and another column with the corresponding number of predicted cells in each image. However, in the prediction section, when using predictions.update.resolve() the images are not printed in the order of test_paths and therefore, neither are the predictions. How could I modify the code so that update.resolve() does not randomise the printing of the images during the prediction but follows the order of test_paths? thank you!

**Comment by BenjaminMidtvedt** [comment]
If the code has not changed too much from the source, you can do something like

```python
test_set = [test_dataset.update().resolve(filename=filename) for filename in test_paths]
```

You can call pipelines with keyword arguments. These will propagate through your pipeline and set any property of that name to have that value. Further up we have 

```python
test_image = dt.LoadImage(
    filename=test_filename_iterator,
    path=lambda filename: os.path.join(DATASET_PATH, "images", filename) + ".tif"
)

test_mask = dt.LoadImage(
    filename=test_image.filename,
    path=lambda filename: os.path.join(DATASET_PATH, "masks", filename) + ".png"
)
```

both these filenames will now be set directly.
